migrations/versions/1200_non_nullable_framework_components_and_.py

```diff
@@ -0,0 +1,50 @@
+"""Non-nullable framework components and enforce at least one True with a constraint
+
+Revision ID: 1200
+Revises: 1190
+Create Date: 2018-05-18 14:55:37.248172
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.sql import table, column
+
+
+# revision identifiers, used by Alembic.
+revision = '1200'
+down_revision = '1190'
+
+
+def upgrade():
+    frameworks = table('frameworks',
+                       column('framework', sa.String()),
+                       column('has_direct_award', sa.BOOLEAN()),
+                       column('has_further_competition', sa.BOOLEAN()))
+
+    # Need to manually insert values for these columns so that the test database gets populated corrected.
+    # Live environments will be populated manually via the API before this migration goes in.
+    op.execute(frameworks.update().where(frameworks.c.framework == 'g-cloud').values(
+        has_direct_award=True,
+        has_further_competition=False,
+    ))
+    op.execute(frameworks.update().where(frameworks.c.framework == 'digital-outcomes-and-specialists').values(
+        has_direct_award=False,
+        has_further_competition=True,
+    ))
```

Comment:
As a general practice it's better to keep data migrations entirely separate to schema migrations, but it probably matters less in postgres than it did in MySQL, and I can't see any actual problems from these staying as they are.